### DISCUSSION TITLE
nuxt的新手，想知道怎么才能正确的部署到服务器上

### DISCUSSION THREAD

**Discussion by heardlyapp** [nuxt/nuxt#8011] [General]
我用nuxt在本地环境开发的一个网站，打包之后传到线上后发现页面源代码不利于seo优化，现在的做法是获取动态路由然后打包成静态页面，每次都要上传30000+的静态文件资源
![1599126435(1)](https://user-images.githubusercontent.com/13566006/92099895-a943cd80-ee0d-11ea-8516-976a343189db.jpg)
我觉得这种做法肯定不正确，但是看了很多文档还是不会，正确的部署方式应该是怎样的

**Comment by danielroe** [comment]
I realise that is a lot of files :slightly_smiling_face: The reason we do that is that by pre-rendering them, your time-to-first-byte is as low as possible - and yes, that is an SEO-friendly approach.

To improve things, if you aren't already doing it:
 * you can register a fallback page at `200.html` that will handle new ids that weren't present when your site was exported
 * you can set up your deployment pipeline to automatically redeploy when your CMS changes the content

Is there a reason you are concerned this might not be SEO-friendly?

**Reply by heardlyapp** [reply]
我查了些资料，有人说可以项目部署到服务器上运行，有这方面的介绍吗

**Reply by danielroe** [reply]
You can find some guides here in the Nuxt FAQs - first one is https://nuxtjs.org/faq/deployment-aws-s3-cloudfront and you should be able to see the rest on the left hand side. Some are for static deployment, and others use a server.
